homeassistant/components/upnp/__init__.py

```diff
@@ -31,7 +30,7 @@
 
 
 REQUIREMENTS = ['async-upnp-client==0.12.4']
-DEPENDENCIES = ['http']
+DEPENDENCIES = ['http', 'discovery']
```

Comment:
I'm actually thinking, why is it that we need both http and discovery to begin with?

Reply:
`discovery` is required as the device is detected via upnp/M-SEARCH.
`http` is required to determine the port home assistant is listening on (really only used the port mapping is enabled.)

Reply:
If we get a SSDP response `discovery` is enabled already. The original component already (only) worked with discovery, but used its own library for it. Now the component is using more existing libraries, such as `netdisco` and `async_upnp_client` (used also by the `dlna_dmr`-component).

The 'externals' (config, workings from the perspective of the user, etc) are mostly kept the same compared to the original `upnp` component. Thus, depending on the `http` component for getting the `hass`-port for port mapping. `discovery` is required as the user has (as before) no way to manually enter a SSDP description URL. Learned from the `dlna_dmr` component, finding this URL is hard for the average user and discovery is preferable.

I get your point and I'm fine with not requiring these, but then there should be:
- a message in the logs clearly stating that discovery is required otherwise it will not work (as before)
- a message in the docs clearly stating that discovery is required otherwise it will not work

Reply:
Philips Hue will, during their config flow, do their own discovery to find the device. The upnp component should do the same.

Reply:
It serves as purpose to find config entries. However, manual config entries will have to rely on their own scanning logic.